rpc/send.go

```diff
@@ -32,6 +32,11 @@ import (
 	"github.com/gogo/protobuf/proto"
 )
 
+// TODO(pmattis): This code is only used by kv.DistSender. Perhaps move it to
+// the kv package. We might also want to de-generalize it so that it knows
```

Comment:
@tamird Would de-generalizing this code so that it only knows about `Batch{Request,Response}` make the move to gRPC easier?